### PR TITLE
feat: make a loop for empirical runs - closes #57

### LINKED ISSUE #57: Make a loop for empirical runs
We need to:
- [x] make a loop over X blocks (5M?) to run both `inspect-many` and `analyze-profit`
- [x] we will need to save the CSVs of the profit from docker to local each time
- [x] then we need to extract the metrics function in another function to run at the end on all csvs

### PR DESCRIPTION
**Pull request by IlluvatarEru**
Signed-off-by: Arthurim <arthurbdauphine@gmail.com>

## What does this PR do?

- Separation of the computation of usd profit and analysis of it into 2 distinctive functions
- For the empirical analysis: loop and alternate the inspection and the profit computation, then do the analysis of the profitt at the end

## Related issue

- [x] Resolves #57 

## Testing

Ran on local on 50k blocks.

## Checklist before merging
- [x] Read the [contributing guide](https://github.com/flashbots/mev-inspect-py/blob/main/CONTRIBUTING.md)
- [x] Installed and ran pre-commit hooks
- [x] All tests pass with `./mev test`

### CODE REVIEW

File: scripts/launch_analysis.sh [hunk 1]
```diff
@@ -4,20 +4,36 @@
...
 blockTo=$((blockFrom+nBlocks))
+window=$((1000))
+reps=$(((${blockTo}-${blockFrom})/${window}))
 startDate=$(date +%s)
 startDateFormatted=$(date +"%Y-%m-%d-%H:%M:%S.")$((startDate))
 echo "Starting analysis from block=${blockFrom} for ${nBlocks} blocks on the ${startDateFormatted}"
-./mev inspect-many $blockFrom $blockTo
-./mev analyze-profit $blockFrom $blockTo True
+echo "${reps}"
+for i in $(seq 0 1 $reps)
```
Comment (IlluvatarEru) [inline scripts/launch_analysis.sh:17]: Can we and should we find a way to run these loops in //? @0xpanoramix

File: scripts/launch_analysis.sh [hunk 2]
```diff
@@ -6,18 +6,33 @@
...
+  echo "to=  $to"
+  echo "$fname"
+  ./mev inspect-many $from $to
+  ./mev compute-usd-profit $from $to True
+done
+./mev analyze-profits $reps True
+declare -a file_names=("total_usd_profit.csv" "top_tokens.csv" "analyze_profit_failures.csv" "profit_distribution.png")
+for fname in "${file_names[@]}"
+do
+  kubectl cp $mevInspectPoolId:resources/$fname ${fname/.csv/_$i.csv};
```
Comment (IlluvatarEru) [inline scripts/launch_analysis.sh]: @0xpanoramix  Do you know if there is a way to clean all csvs in the kube/docker envt? WOuld be great to delete them all before running the script each time.

File: profit_analysis/analysis.py [hunk 3]
```diff
@@ -37,6 +38,52 @@
...
+            PRICE_RECEIVED_KEY,
+            AMOUNT_DEBT_KEY,
+            TOKEN_DEBT_KEY,
+            PRICE_DEBT_KEY,
+            PROFIT_USD_KEY,
+            CATEGORY_KEY,
+        ]
+    )
+    for i in range(n):
+        profit = pd.read_csv(DATA_PATH + USD_PROFIT_FILE_NAME + f"_{i}.csv")
```
Comment (0xpanoramix) [inline profit_analysis/analysis.py]: I think you can also use the following, which is easier to read:

```suggestion
        profit = pd.read_csv(f"{DATA_PATH}{USD_PROFIT_FILE_NAME}_{i}.csv")
```

File: scripts/launch_analysis.sh (hunk 1 above)
Comment (0xpanoramix) [reply scripts/launch_analysis.sh:17]: We can and yes we should, but let's do this in another PR.

**Review by 0xpanoramix — CHANGES_REQUESTED** [review]

**Review by 0xpanoramix — APPROVED** [review]

**Pull request merged**